{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/FarahYusri12/Tutorial_Reproduce_Genomic_paper_Figures/blob/main/Allignment_to_the_genome.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4-79auA8h8Vt",
        "outputId": "75737b1f-730f-43e6-92f5-db9fa1f90cf1"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/genome\n",
            "/content/genome\n"
          ]
        }
      ],
      "source": [
        "!mkdir genome  #make new folder\n",
        "%cd genome\n",
        "!pwd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mWeRa2kpsYqF",
        "outputId": "005d98bf-35b6-4711-9f2d-a93a599aff55"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 https://cli.github.com/packages stable InRelease\n",
            "\r0% [Connecting to archive.ubuntu.com (91.189.92.24)] [Connecting to security.ub\r                                                                               \rGet:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "\r0% [Connecting to archive.ubuntu.com (91.189.92.24)] [Connecting to security.ub\r                                                                               \rGet:3 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "Get:4 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:5 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [2,201 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:9 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ Packages [83.6 kB]\n",
            "Get:10 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,842 kB]\n",
            "Get:11 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,510 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:13 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,571 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,598 kB]\n",
            "Get:15 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease [18.1 kB]\n",
            "Hit:16 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:17 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:18 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [6,081 kB]\n",
            "Get:19 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 Packages [38.5 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [6,281 kB]\n",
            "Get:21 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,286 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,901 kB]\n",
            "Fetched 37.8 MB in 4s (8,502 kB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "Suggested packages:\n",
            "  bowtie2-examples\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2\n",
            "0 upgraded, 1 newly installed, 0 to remove and 66 not upgraded.\n",
            "Need to get 1,169 kB of archives.\n",
            "After this operation, 3,735 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bowtie2 amd64 2.4.4-1 [1,169 kB]\n",
            "Fetched 1,169 kB in 2s (698 kB/s)\n",
            "Selecting previously unselected package bowtie2.\n",
            "(Reading database ... 121713 files and directories currently installed.)\n",
            "Preparing to unpack .../bowtie2_2.4.4-1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.4.4-1) ...\n",
            "Setting up bowtie2 (2.4.4-1) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get update\n",
        "!apt-get install  -y bowtie2  #install bowtie2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://genome-idx.s3.amazonaws.com/bt/GRCh38_noalt_as.zip\n",
        "#download pre-build index for bowtie2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YhFTowiJTLj-",
        "outputId": "97c49e46-fd71-42c8-fc1d-05ed8170516a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-12-08 06:32:19--  https://genome-idx.s3.amazonaws.com/bt/GRCh38_noalt_as.zip\n",
            "Resolving genome-idx.s3.amazonaws.com (genome-idx.s3.amazonaws.com)... 54.231.228.201, 16.15.185.112, 3.5.30.209, ...\n",
            "Connecting to genome-idx.s3.amazonaws.com (genome-idx.s3.amazonaws.com)|54.231.228.201|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3749239215 (3.5G) [application/zip]\n",
            "Saving to: ‘GRCh38_noalt_as.zip’\n",
            "\n",
            "GRCh38_noalt_as.zip 100%[===================>]   3.49G  41.3MB/s    in 92s     \n",
            "\n",
            "2025-12-08 06:33:51 (39.0 MB/s) - ‘GRCh38_noalt_as.zip’ saved [3749239215/3749239215]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip GRCh38_noalt_as.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_HsvotUgTrPI",
        "outputId": "3d281231-c2f3-4d7d-8250-11597b68eb68"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  GRCh38_noalt_as.zip\n",
            "   creating: GRCh38_noalt_as/\n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.3.bt2  \n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.rev.2.bt2  \n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.rev.1.bt2  \n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.1.bt2  \n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.2.bt2  \n",
            "  inflating: GRCh38_noalt_as/GRCh38_noalt_as.4.bt2  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-0P-9q9rppmt",
        "outputId": "175c9505-2b99-4e3a-c235-6abf11797cd7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp -r /content/genome /content/drive/MyDrive/portfolio"
      ],
      "metadata": {
        "id": "JMIhSiQAoyf6"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "!bowtie2 -x /content/drive/MyDrive/portfolio/genome/GRCh38_noalt_as/GRCh38_noalt_as \\\n",
        "-U \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810900.fastq.gz\" \\\n",
        "-S /content/SRR1810900.sam \\\n",
        "--threads 4 \\\n",
        "-k 1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BvPpgX6ppLQi",
        "outputId": "dd7b5aca-03e0-4c35-ddbc-2f4803142ca4"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "24549590 reads; of these:\n",
            "  24549590 (100.00%) were unpaired; of these:\n",
            "    895212 (3.65%) aligned 0 times\n",
            "    23654378 (96.35%) aligned exactly 1 time\n",
            "    0 (0.00%) aligned >1 times\n",
            "96.35% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#install samtools\n",
        "!apt-get update -qq\n",
        "!apt-get -y install samtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "u-nTF-BMz7OK",
        "outputId": "47340b75-2948-4813-ce53-f6328fec4fec"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libhtscodecs2\n",
            "Suggested packages:\n",
            "  cwltool\n",
            "The following NEW packages will be installed:\n",
            "  libhts3 libhtscodecs2 samtools\n",
            "0 upgraded, 3 newly installed, 0 to remove and 66 not upgraded.\n",
            "Need to get 963 kB of archives.\n",
            "After this operation, 2,270 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Fetched 963 kB in 1s (937 kB/s)\n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 121739 files and directories currently installed.)\n",
            "Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -bS \"/content/SRR1810900.sam\" > SRR1810900.bam && \\\n",
        "samtools sort SRR1810900.bam -o SRR1810900.sorted.bam && \\\n",
        "samtools index SRR1810900.sorted.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ekiwUd-a20Ye",
        "outputId": "4fab08fe-6860-49ec-a6dc-dd8ee6082f12"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 6 files and 1 in-memory blocks...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "!bowtie2 -x /content/drive/MyDrive/portfolio/genome/GRCh38_noalt_as/GRCh38_noalt_as \\\n",
        "-U \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810907.fastq.gz\" \\\n",
        "-S /content/SRR1810907.sam \\\n",
        "--threads 4 \\\n",
        "-k 1"
      ],
      "metadata": {
        "id": "TlDVCvTd6krJ",
        "outputId": "abf809ba-4ef6-4000-e312-b4e0abced206",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 12,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "27521260 reads; of these:\n",
            "  27521260 (100.00%) were unpaired; of these:\n",
            "    731406 (2.66%) aligned 0 times\n",
            "    26789854 (97.34%) aligned exactly 1 time\n",
            "    0 (0.00%) aligned >1 times\n",
            "97.34% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -bS \"/content/SRR1810907.sam\" > SRR1810907.bam && \\\n",
        "samtools sort SRR1810907.bam -o SRR1810907.sorted.bam && \\\n",
        "samtools index SRR1810907.sorted.bam"
      ],
      "metadata": {
        "id": "DeGIaTOcDrBt",
        "outputId": "f6cadc99-c679-4eea-cffe-fdb75e27b99b",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR1810900.sorted.bam \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "7Z_Rwx8DFpMn"
      },
      "execution_count": 27,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR1810900.sorted.bam.bai \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "yyr3TwNBF5LP"
      },
      "execution_count": 28,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR1810907.sorted.bam \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "f_S6rzUzGA7Q"
      },
      "execution_count": 32,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR1810907.sorted.bam.bai \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "0xnRsADxGBl7"
      },
      "execution_count": 33,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -H /content/SRR1810900.sam | head\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AE30A2gvH4JO",
        "outputId": "4c450adf-8546-4d0d-fc74-523b28c80f0e"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@HD\tVN:1.0\tSO:unsorted\n",
            "@SQ\tSN:chr1\tLN:248956422\n",
            "@SQ\tSN:chr2\tLN:242193529\n",
            "@SQ\tSN:chr3\tLN:198295559\n",
            "@SQ\tSN:chr4\tLN:190214555\n",
            "@SQ\tSN:chr5\tLN:181538259\n",
            "@SQ\tSN:chr6\tLN:170805979\n",
            "@SQ\tSN:chr7\tLN:159345973\n",
            "@SQ\tSN:chr8\tLN:145138636\n",
            "@SQ\tSN:chr9\tLN:138394717\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1️⃣ Convert SAM → BAM\n",
        "!samtools view -bS SRR1810900.sam -o SRR1810900.bam\n",
        "\n",
        "# 2️⃣ Sort BAM by coordinate\n",
        "!samtools sort SRR1810900.bam -o SRR1810900.sorted.bam\n",
        "\n",
        "# 3️⃣ Index BAM\n",
        "!samtools index SRR1810900.sorted.bam\n",
        "\n",
        "# 4️⃣ Check mapping stats\n",
        "!samtools flagstat SRR1810900.sorted.bam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PNeolhjnIPBX",
        "outputId": "8a52139a-3e95-484c-b8e8-b4805b4472f1"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 6 files and 1 in-memory blocks...\n",
            "24549590 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "24549590 + 0 primary\n",
            "0 + 0 secondary\n",
            "0 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "23654378 + 0 mapped (96.35% : N/A)\n",
            "23654378 + 0 primary mapped (96.35% : N/A)\n",
            "0 + 0 paired in sequencing\n",
            "0 + 0 read1\n",
            "0 + 0 read2\n",
            "0 + 0 properly paired (N/A : N/A)\n",
            "0 + 0 with itself and mate mapped\n",
            "0 + 0 singletons (N/A : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -H /content/SRR1810907.sam | head\n"
      ],
      "metadata": {
        "id": "CvbI-qD3RyGL",
        "outputId": "0bea6cb1-bf27-4745-a8e4-7c3ee0ca2a2c",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 30,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "@HD\tVN:1.0\tSO:unsorted\n",
            "@SQ\tSN:chr1\tLN:248956422\n",
            "@SQ\tSN:chr2\tLN:242193529\n",
            "@SQ\tSN:chr3\tLN:198295559\n",
            "@SQ\tSN:chr4\tLN:190214555\n",
            "@SQ\tSN:chr5\tLN:181538259\n",
            "@SQ\tSN:chr6\tLN:170805979\n",
            "@SQ\tSN:chr7\tLN:159345973\n",
            "@SQ\tSN:chr8\tLN:145138636\n",
            "@SQ\tSN:chr9\tLN:138394717\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1️⃣ Convert SAM → BAM\n",
        "!samtools view -bS SRR1810907.sam -o SRR1810907.bam\n",
        "\n",
        "# 2️⃣ Sort BAM by coordinate\n",
        "!samtools sort SRR1810907.bam -o SRR1810907.sorted.bam\n",
        "\n",
        "# 3️⃣ Index BAM\n",
        "!samtools index SRR1810907.sorted.bam\n",
        "\n",
        "# 4️⃣ Check mapping stats\n",
        "!samtools flagstat SRR1810907.sorted.bam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AnOXFLA1R7CI",
        "outputId": "91e9470d-307d-4111-8043-158d72d77664"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 7 files and 1 in-memory blocks...\n",
            "27521260 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "27521260 + 0 primary\n",
            "0 + 0 secondary\n",
            "0 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "26789854 + 0 mapped (97.34% : N/A)\n",
            "26789854 + 0 primary mapped (97.34% : N/A)\n",
            "0 + 0 paired in sequencing\n",
            "0 + 0 read1\n",
            "0 + 0 read2\n",
            "0 + 0 properly paired (N/A : N/A)\n",
            "0 + 0 with itself and mate mapped\n",
            "0 + 0 singletons (N/A : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "!bowtie2 -x /content/drive/MyDrive/portfolio/genome/GRCh38_noalt_as/GRCh38_noalt_as \\\n",
        "-U \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810912.fastq.gz\" \\\n",
        "-S /content/SRR1810912.sam \\\n",
        "--threads 4 \\\n",
        "-k 1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UmzcN4bhUXc6",
        "outputId": "a2f50d11-79d0-4682-eb4d-7f7ef4373e82"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "31246182 reads; of these:\n",
            "  31246182 (100.00%) were unpaired; of these:\n",
            "    1011695 (3.24%) aligned 0 times\n",
            "    30234487 (96.76%) aligned exactly 1 time\n",
            "    0 (0.00%) aligned >1 times\n",
            "96.76% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools quickcheck -v SRR1810918.sorted.bam\n"
      ],
      "metadata": {
        "id": "pGNX2qgMflJT"
      },
      "execution_count": 41,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Convert SAM → BAM\n",
        "!samtools view -bS SRR1810912.sam -o SRR1810912.bam\n",
        "\n",
        "# 2️. Sort BAM by coordinate\n",
        "!samtools sort SRR1810912.bam -o SRR1810912.sorted.bam\n",
        "\n",
        "# 3️. Index BAM\n",
        "!samtools index SRR1810912.sorted.bam\n",
        "\n",
        "# . Check mapping stats\n",
        "!samtools flagstat SRR1810912.sorted.bam\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cQGryHRlfz2u",
        "outputId": "c55b372f-959a-4bc8-8ed6-f3f8da814e3d"
      },
      "execution_count": 37,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 8 files and 1 in-memory blocks...\n",
            "31246182 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "31246182 + 0 primary\n",
            "0 + 0 secondary\n",
            "0 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "30234487 + 0 mapped (96.76% : N/A)\n",
            "30234487 + 0 primary mapped (96.76% : N/A)\n",
            "0 + 0 paired in sequencing\n",
            "0 + 0 read1\n",
            "0 + 0 read2\n",
            "0 + 0 properly paired (N/A : N/A)\n",
            "0 + 0 with itself and mate mapped\n",
            "0 + 0 singletons (N/A : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "!bowtie2 -x /content/drive/MyDrive/portfolio/genome/GRCh38_noalt_as/GRCh38_noalt_as \\\n",
        "-U \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810918.fastq.gz\" \\\n",
        "-S /content/SRR1810918.sam \\\n",
        "--threads 4 \\\n",
        "-k 1"
      ],
      "metadata": {
        "id": "d-LBTM0FisHX",
        "outputId": "6dbc6421-4a4d-42e7-8120-4275efb05d0c",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 39,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "34776462 reads; of these:\n",
            "  34776462 (100.00%) were unpaired; of these:\n",
            "    443610 (1.28%) aligned 0 times\n",
            "    34332852 (98.72%) aligned exactly 1 time\n",
            "    0 (0.00%) aligned >1 times\n",
            "98.72% overall alignment rate\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Convert SAM → BAM\n",
        "!samtools view -bS SRR1810918.sam -o SRR1810918.bam\n",
        "\n",
        "# 2️. Sort BAM by coordinate\n",
        "!samtools sort SRR1810918.bam -o SRR1810918.sorted.bam\n",
        "\n",
        "# 3️. Index BAM\n",
        "!samtools index SRR1810918.sorted.bam\n",
        "\n",
        "# 4. Check mapping stats\n",
        "!samtools flagstat SRR1810918.sorted.bam\n"
      ],
      "metadata": {
        "id": "B4z2YO6Et4h8",
        "outputId": "2bdfc56f-dd66-423a-bfb1-54e6f45150d3",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 9 files and 1 in-memory blocks...\n",
            "34776462 + 0 in total (QC-passed reads + QC-failed reads)\n",
            "34776462 + 0 primary\n",
            "0 + 0 secondary\n",
            "0 + 0 supplementary\n",
            "0 + 0 duplicates\n",
            "0 + 0 primary duplicates\n",
            "34332852 + 0 mapped (98.72% : N/A)\n",
            "34332852 + 0 primary mapped (98.72% : N/A)\n",
            "0 + 0 paired in sequencing\n",
            "0 + 0 read1\n",
            "0 + 0 read2\n",
            "0 + 0 properly paired (N/A : N/A)\n",
            "0 + 0 with itself and mate mapped\n",
            "0 + 0 singletons (N/A : N/A)\n",
            "0 + 0 with mate mapped to a different chr\n",
            "0 + 0 with mate mapped to a different chr (mapQ>=5)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!cp /content/SRR1810912.sorted.bam \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp /content/SRR1810912.sorted.bam.bai \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp /content/SRR1810918.sorted.bam \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp /content/SRR1810918.sorted.bam.bai \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "UWz4CFQcxD1D"
      },
      "execution_count": 42,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "mount_file_id": "https://github.com/FarahYusri12/Tutorial_Reproduce_Genomic_paper_Figures/blob/main/Allignment_to_the_genome.ipynb",
      "authorship_tag": "ABX9TyPLaiaNsLKkkrzzfV/cg9TY",
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}