{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "mount_file_id": "1aUjV3GE27WUBeTKECleXIMrOaqGwZHuX",
      "authorship_tag": "ABX9TyOPJqTcy4OWxCuLjzxQSupl",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/FarahYusri12/Tutorial_Reproduce_Genomic_paper_Figures/blob/main/Generate_bigwig_files_and_peak_calling.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mdpJNesF4d0-",
        "outputId": "05ec9f5c-3b36-411c-eeac-2389bf07a5fe"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting deeptools\n",
            "  Downloading deepTools-3.5.6-py3-none-any.whl.metadata (4.0 kB)\n",
            "Requirement already satisfied: numpy>=2.0.0 in /usr/local/lib/python3.12/dist-packages (from deeptools) (2.0.2)\n",
            "Requirement already satisfied: scipy>=0.17.0 in /usr/local/lib/python3.12/dist-packages (from deeptools) (1.16.3)\n",
            "Requirement already satisfied: matplotlib>=3.5.0 in /usr/local/lib/python3.12/dist-packages (from deeptools) (3.10.0)\n",
            "Collecting pysam>=0.14.0 (from deeptools)\n",
            "  Downloading pysam-0.23.3-cp312-cp312-manylinux_2_28_x86_64.whl.metadata (1.7 kB)\n",
            "Collecting numpydoc>=0.5 (from deeptools)\n",
            "  Downloading numpydoc-1.10.0-py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting pyBigWig>=0.2.1 (from deeptools)\n",
            "  Downloading pyBigWig-0.3.24-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (19 kB)\n",
            "Collecting py2bit>=0.2.0 (from deeptools)\n",
            "  Downloading py2bit-0.3.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.0 kB)\n",
            "Requirement already satisfied: plotly>=4.9 in /usr/local/lib/python3.12/dist-packages (from deeptools) (5.24.1)\n",
            "Collecting deeptoolsintervals>=0.1.8 (from deeptools)\n",
            "  Downloading deeptoolsintervals-0.1.9.tar.gz (41 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m41.1/41.1 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (1.3.3)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (4.61.0)\n",
            "Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (1.4.9)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (25.0)\n",
            "Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (11.3.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (3.2.5)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.12/dist-packages (from matplotlib>=3.5.0->deeptools) (2.9.0.post0)\n",
            "Requirement already satisfied: sphinx>=6 in /usr/local/lib/python3.12/dist-packages (from numpydoc>=0.5->deeptools) (8.2.3)\n",
            "Requirement already satisfied: tenacity>=6.2.0 in /usr/local/lib/python3.12/dist-packages (from plotly>=4.9->deeptools) (9.1.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.7->matplotlib>=3.5.0->deeptools) (1.17.0)\n",
            "Requirement already satisfied: sphinxcontrib-applehelp>=1.0.7 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n",
            "Requirement already satisfied: sphinxcontrib-devhelp>=1.0.6 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n",
            "Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.6 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.1.0)\n",
            "Requirement already satisfied: sphinxcontrib-jsmath>=1.0.1 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.0.1)\n",
            "Requirement already satisfied: sphinxcontrib-qthelp>=1.0.6 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n",
            "Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.9 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.0.0)\n",
            "Requirement already satisfied: Jinja2>=3.1 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (3.1.6)\n",
            "Requirement already satisfied: Pygments>=2.17 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.19.2)\n",
            "Requirement already satisfied: docutils<0.22,>=0.20 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (0.21.2)\n",
            "Requirement already satisfied: snowballstemmer>=2.2 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (3.0.1)\n",
            "Requirement already satisfied: babel>=2.13 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.17.0)\n",
            "Requirement already satisfied: alabaster>=0.7.14 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.0.0)\n",
            "Requirement already satisfied: imagesize>=1.3 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (1.4.1)\n",
            "Requirement already satisfied: requests>=2.30.0 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (2.32.4)\n",
            "Requirement already satisfied: roman-numerals-py>=1.0.0 in /usr/local/lib/python3.12/dist-packages (from sphinx>=6->numpydoc>=0.5->deeptools) (3.1.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.12/dist-packages (from Jinja2>=3.1->sphinx>=6->numpydoc>=0.5->deeptools) (3.0.3)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests>=2.30.0->sphinx>=6->numpydoc>=0.5->deeptools) (3.4.4)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests>=2.30.0->sphinx>=6->numpydoc>=0.5->deeptools) (3.11)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests>=2.30.0->sphinx>=6->numpydoc>=0.5->deeptools) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests>=2.30.0->sphinx>=6->numpydoc>=0.5->deeptools) (2025.11.12)\n",
            "Downloading deepTools-3.5.6-py3-none-any.whl (208 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m208.8/208.8 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading numpydoc-1.10.0-py3-none-any.whl (69 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m69.3/69.3 kB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading py2bit-0.3.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (50 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.1/50.1 kB\u001b[0m \u001b[31m1.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyBigWig-0.3.24-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (187 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m187.1/187.1 kB\u001b[0m \u001b[31m5.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pysam-0.23.3-cp312-cp312-manylinux_2_28_x86_64.whl (24.0 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m24.0/24.0 MB\u001b[0m \u001b[31m23.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: deeptoolsintervals\n",
            "  Building wheel for deeptoolsintervals (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for deeptoolsintervals: filename=deeptoolsintervals-0.1.9-cp312-cp312-linux_x86_64.whl size=113640 sha256=07b43d2192c6d2977f00d687a76791b3fe26501fd14662ccf2f5bf98622868ab\n",
            "  Stored in directory: /root/.cache/pip/wheels/b3/24/1a/4ee9c2f3a77c10767cda723c6e4c93345e0379e769d002bb94\n",
            "Successfully built deeptoolsintervals\n",
            "Installing collected packages: py2bit, deeptoolsintervals, pysam, pyBigWig, numpydoc, deeptools\n",
            "Successfully installed deeptools-3.5.6 deeptoolsintervals-0.1.9 numpydoc-1.10.0 py2bit-0.3.3 pyBigWig-0.3.24 pysam-0.23.3\n"
          ]
        }
      ],
      "source": [
        "#install deeptolls\n",
        "!pip install deeptools"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#generate bigwig file (SRR1810900)\n",
        "!bamCoverage \\\n",
        "--bam \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810900.sorted.bam\" \\\n",
        "-o SRR1810900.bw \\\n",
        "--normalizeUsing RPKM \\\n",
        "--extendReads 200"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3VE-Yo5B7L_7",
        "outputId": "c057f532-0946-4dda-dcf8-43999676dcea"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "normalization: RPKM\n",
            "bamFilesList: ['/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810900.sorted.bam']\n",
            "binLength: 50\n",
            "numberOfSamples: None\n",
            "blackListFileName: None\n",
            "skipZeroOverZero: False\n",
            "bed_and_bin: False\n",
            "genomeChunkSize: None\n",
            "defaultFragmentLength: 200\n",
            "numberOfProcessors: 1\n",
            "verbose: False\n",
            "region: None\n",
            "bedFile: None\n",
            "minMappingQuality: None\n",
            "ignoreDuplicates: False\n",
            "chrsToSkip: []\n",
            "stepSize: 50\n",
            "center_read: False\n",
            "samFlag_include: None\n",
            "samFlag_exclude: None\n",
            "minFragmentLength: 0\n",
            "maxFragmentLength: 0\n",
            "zerosToNans: False\n",
            "smoothLength: None\n",
            "save_data: False\n",
            "out_file_for_raw_data: None\n",
            "maxPairedFragmentLength: 800\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# mount files fron drive to content\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810907.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810907.sorted.bam.bai\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810912.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810912.sorted.bam.bai\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810918.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810918.sorted.bam.bai\" ."
      ],
      "metadata": {
        "id": "k-LxAFQLAL69"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# generate bigwig files (loop)\n",
        "%%bash\n",
        "for bam in *.sorted.bam; do\n",
        "out=${bam/.sorted.bam}.bw\n",
        "bamCoverage \\\n",
        "--bam \"$bam\" \\\n",
        "--normalizeUsing RPKM \\\n",
        "--extendReads 200 \\\n",
        "-o \"$out\"\n",
        "done"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BsrqnUgAFIA4",
        "outputId": "dddba236-0931-491b-d0d8-8f7d3ae78877"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "normalization: RPKM\n",
            "bamFilesList: ['SRR1810907.sorted.bam']\n",
            "binLength: 50\n",
            "numberOfSamples: None\n",
            "blackListFileName: None\n",
            "skipZeroOverZero: False\n",
            "bed_and_bin: False\n",
            "genomeChunkSize: None\n",
            "defaultFragmentLength: 200\n",
            "numberOfProcessors: 1\n",
            "verbose: False\n",
            "region: None\n",
            "bedFile: None\n",
            "minMappingQuality: None\n",
            "ignoreDuplicates: False\n",
            "chrsToSkip: []\n",
            "stepSize: 50\n",
            "center_read: False\n",
            "samFlag_include: None\n",
            "samFlag_exclude: None\n",
            "minFragmentLength: 0\n",
            "maxFragmentLength: 0\n",
            "zerosToNans: False\n",
            "smoothLength: None\n",
            "save_data: False\n",
            "out_file_for_raw_data: None\n",
            "maxPairedFragmentLength: 800\n",
            "normalization: RPKM\n",
            "bamFilesList: ['SRR1810912.sorted.bam']\n",
            "binLength: 50\n",
            "numberOfSamples: None\n",
            "blackListFileName: None\n",
            "skipZeroOverZero: False\n",
            "bed_and_bin: False\n",
            "genomeChunkSize: None\n",
            "defaultFragmentLength: 200\n",
            "numberOfProcessors: 1\n",
            "verbose: False\n",
            "region: None\n",
            "bedFile: None\n",
            "minMappingQuality: None\n",
            "ignoreDuplicates: False\n",
            "chrsToSkip: []\n",
            "stepSize: 50\n",
            "center_read: False\n",
            "samFlag_include: None\n",
            "samFlag_exclude: None\n",
            "minFragmentLength: 0\n",
            "maxFragmentLength: 0\n",
            "zerosToNans: False\n",
            "smoothLength: None\n",
            "save_data: False\n",
            "out_file_for_raw_data: None\n",
            "maxPairedFragmentLength: 800\n",
            "normalization: RPKM\n",
            "bamFilesList: ['SRR1810918.sorted.bam']\n",
            "binLength: 50\n",
            "numberOfSamples: None\n",
            "blackListFileName: None\n",
            "skipZeroOverZero: False\n",
            "bed_and_bin: False\n",
            "genomeChunkSize: None\n",
            "defaultFragmentLength: 200\n",
            "numberOfProcessors: 1\n",
            "verbose: False\n",
            "region: None\n",
            "bedFile: None\n",
            "minMappingQuality: None\n",
            "ignoreDuplicates: False\n",
            "chrsToSkip: []\n",
            "stepSize: 50\n",
            "center_read: False\n",
            "samFlag_include: None\n",
            "samFlag_exclude: None\n",
            "minFragmentLength: 0\n",
            "maxFragmentLength: 0\n",
            "zerosToNans: False\n",
            "smoothLength: None\n",
            "save_data: False\n",
            "out_file_for_raw_data: None\n",
            "maxPairedFragmentLength: 800\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#mount bw files into drive\n",
        "!cp \"/content/SRR1810900.bw\" \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp \"/content/SRR1810907.bw\" \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp \"/content/SRR1810912.bw\" \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp \"/content/SRR1810918.bw\" \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "2Yxd3-8HNQba"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "peak calling"
      ],
      "metadata": {
        "id": "ifjBsdz3dQ-D"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install --upgrade macs3"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XXD0WccMdP68",
        "outputId": "85a29da5-3108-4637-9617-ee636854a326"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs3\n",
            "  Downloading macs3-3.0.3.tar.gz (702 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m702.1/702.1 kB\u001b[0m \u001b[31m11.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy>=1.25 in /usr/local/lib/python3.12/dist-packages (from macs3) (2.0.2)\n",
            "Requirement already satisfied: scipy>=1.12 in /usr/local/lib/python3.12/dist-packages (from macs3) (1.16.3)\n",
            "Collecting hmmlearn>=0.3.2 (from macs3)\n",
            "  Using cached hmmlearn-0.3.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)\n",
            "Requirement already satisfied: scikit-learn>=1.3 in /usr/local/lib/python3.12/dist-packages (from macs3) (1.6.1)\n",
            "Collecting cykhash>=2.0 (from macs3)\n",
            "  Using cached cykhash-2.0.1-cp312-cp312-linux_x86_64.whl\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.12/dist-packages (from scikit-learn>=1.3->macs3) (1.5.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.12/dist-packages (from scikit-learn>=1.3->macs3) (3.6.0)\n",
            "Using cached hmmlearn-0.3.3-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (165 kB)\n",
            "Building wheels for collected packages: macs3\n",
            "  Building wheel for macs3 (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs3: filename=macs3-3.0.3-cp312-cp312-linux_x86_64.whl size=3665438 sha256=0b95a6d53a6ade932c8ec43da7b21e43a072cda45ecbea7bf5fe0505d2d63b51\n",
            "  Stored in directory: /root/.cache/pip/wheels/c4/3c/ca/661d80b60be21eefdde9e0788a8465c9de499dd1a227c6cf91\n",
            "Successfully built macs3\n",
            "Installing collected packages: cykhash, hmmlearn, macs3\n",
            "Successfully installed cykhash-2.0.1 hmmlearn-0.3.3 macs3-3.0.3\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#mount bam file to contenr\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810900.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810900.sorted.bam.bai\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810907.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810907.sorted.bam.bai\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810912.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810912.sorted.bam.bai\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810918.sorted.bam\" .\n",
        "!cp \"/content/drive/MyDrive/portfolio/Chip-seq tutorial/SRR1810918.sorted.bam.bai\" ."
      ],
      "metadata": {
        "id": "_X1EAAlofkxv"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#SRR1810900(treatment), SRR1810912(control)\n",
        "%%bash\n",
        "macs3 callpeak \\\n",
        "-t SRR1810900.sorted.bam \\\n",
        "-c SRR1810912.sorted.bam \\\n",
        "-f BAM \\\n",
        "-g hs \\\n",
        "-n SRR1810900 \\\n",
        "--outdir SRR1810900_peak \\\n",
        "--keep-dup all \\\n",
        "--call-summits"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LpMItRVGhtUo",
        "outputId": "338c5c1a-b50e-4ea0-a534-f7b69d138062"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "INFO  @ 12 Dec 2025 17:14:57: [34 MB] \n",
            "# Command line: callpeak -t SRR1810900.sorted.bam -c SRR1810912.sorted.bam -f BAM -g hs -n SRR1810900 --outdir SRR1810900_peak --keep-dup all --call-summits\n",
            "# ARGUMENTS LIST:\n",
            "# name = SRR1810900\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['SRR1810900.sorted.bam']\n",
            "# control file = ['SRR1810912.sorted.bam']\n",
            "# effective genome size = 2.91e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff = 5.00e-02\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is off\n",
            "# Paired-End mode is off\n",
            "# Searching for subpeak summits is on\n",
            " \n",
            "INFO  @ 12 Dec 2025 17:14:57: [34 MB] #1 read tag files... \n",
            "INFO  @ 12 Dec 2025 17:14:57: [34 MB] #1 read treatment tags... \n",
            "INFO  @ 12 Dec 2025 17:15:01: [67 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:04: [70 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:08: [72 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:12: [75 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:15: [79 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:18: [85 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:23: [86 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:27: [90 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:30: [97 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:34: [103 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:38: [106 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:41: [107 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:45: [111 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:48: [116 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:51: [121 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:55: [127 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:15:58: [132 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:02: [132 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:05: [135 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:09: [140 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:13: [146 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:16: [150 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:19: [154 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:24: [202 MB] 23654378 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:16:24: [202 MB] #1.2 read input tags... \n",
            "INFO  @ 12 Dec 2025 17:16:27: [207 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:30: [213 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:34: [224 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:38: [227 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:41: [232 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:44: [232 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:47: [235 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:52: [243 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:55: [243 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:16:58: [245 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:01: [251 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:06: [259 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:09: [260 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:14: [262 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:19: [266 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:22: [269 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:25: [274 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:28: [279 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:33: [283 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:36: [286 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:40: [291 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:43: [297 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:47: [303 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:51: [306 MB]  24000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:54: [310 MB]  25000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:17:57: [314 MB]  26000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:18:01: [318 MB]  27000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:18:05: [324 MB]  28000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:18:08: [325 MB]  29000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:18:11: [329 MB]  30000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:18:14: [373 MB] 30234487 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #1 tag size is determined as 50 bps \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #1 tag size = 50.0 \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #1  total tags in treatment: 23654378 \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #1  total tags in control: 30234487 \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #1 finished! \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #2 Build Peak Model... \n",
            "INFO  @ 12 Dec 2025 17:18:15: [373 MB] #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ 12 Dec 2025 17:18:26: [387 MB] #2 Total number of paired peaks: 7678 \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #2 Model building with cross-correlation: Done \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #2 finished! \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #2 predicted fragment length is 196 bps \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #2 alternative fragment length(s) may be 196 bps \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #2.2 Generate R script for model : SRR1810900_peak/SRR1810900_model.r \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #3 Call peaks... \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #3 Going to call summits inside each peak ... \n",
            "INFO  @ 12 Dec 2025 17:18:31: [387 MB] #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ 12 Dec 2025 17:26:35: [1339 MB] #3 Call peaks for each chromosome... \n",
            "INFO  @ 12 Dec 2025 17:28:00: [1339 MB] #4 Write output xls file... SRR1810900_peak/SRR1810900_peaks.xls \n",
            "INFO  @ 12 Dec 2025 17:28:00: [1339 MB] #4 Write peak in narrowPeak format file... SRR1810900_peak/SRR1810900_peaks.narrowPeak \n",
            "INFO  @ 12 Dec 2025 17:28:00: [1339 MB] #4 Write summits bed file... SRR1810900_peak/SRR1810900_summits.bed \n",
            "INFO  @ 12 Dec 2025 17:28:00: [1339 MB] Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#SRR1810907(treatment), SRR1810912(control)\n",
        "%%bash\n",
        "macs3 callpeak \\\n",
        "-t SRR1810907.sorted.bam \\\n",
        "-c SRR1810912.sorted.bam \\\n",
        "-f BAM \\\n",
        "-g hs \\\n",
        "-n SRR1810907 \\\n",
        "--outdir SRR1810907_peak \\\n",
        "--keep-dup all \\\n",
        "--call-summits"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VTqE54enm2e3",
        "outputId": "048729fb-212a-49fb-fe46-a38e6963cf61"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "INFO  @ 12 Dec 2025 17:29:06: [34 MB] \n",
            "# Command line: callpeak -t SRR1810907.sorted.bam -c SRR1810912.sorted.bam -f BAM -g hs -n SRR1810907 --outdir SRR1810907_peak --keep-dup all --call-summits\n",
            "# ARGUMENTS LIST:\n",
            "# name = SRR1810907\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['SRR1810907.sorted.bam']\n",
            "# control file = ['SRR1810912.sorted.bam']\n",
            "# effective genome size = 2.91e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff = 5.00e-02\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is off\n",
            "# Paired-End mode is off\n",
            "# Searching for subpeak summits is on\n",
            " \n",
            "INFO  @ 12 Dec 2025 17:29:06: [34 MB] #1 read tag files... \n",
            "INFO  @ 12 Dec 2025 17:29:06: [34 MB] #1 read treatment tags... \n",
            "INFO  @ 12 Dec 2025 17:29:11: [64 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:14: [67 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:17: [80 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:20: [80 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:25: [82 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:28: [85 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:31: [86 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:34: [96 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:39: [96 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:42: [105 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:45: [108 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:48: [109 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:53: [112 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:56: [117 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:29:59: [123 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:02: [128 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:07: [133 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:10: [138 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:13: [138 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:16: [139 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:20: [144 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:24: [148 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:27: [152 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:30: [155 MB]  24000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:34: [160 MB]  25000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:38: [164 MB]  26000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:42: [212 MB] 26789854 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:30:42: [212 MB] #1.2 read input tags... \n",
            "INFO  @ 12 Dec 2025 17:30:46: [217 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:50: [223 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:53: [234 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:30:57: [237 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:00: [241 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:04: [241 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:08: [245 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:11: [253 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:14: [253 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:18: [256 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:22: [261 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:25: [269 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:28: [270 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:32: [272 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:36: [276 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:39: [279 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:42: [284 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:46: [289 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:50: [293 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:53: [296 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:31:56: [301 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:00: [307 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:04: [312 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:07: [316 MB]  24000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:10: [320 MB]  25000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:13: [324 MB]  26000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:18: [328 MB]  27000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:21: [334 MB]  28000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:24: [334 MB]  29000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:27: [339 MB]  30000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:32:30: [384 MB] 30234487 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #1 tag size is determined as 50 bps \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #1 tag size = 50.0 \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #1  total tags in treatment: 26789854 \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #1  total tags in control: 30234487 \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #1 finished! \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #2 Build Peak Model... \n",
            "INFO  @ 12 Dec 2025 17:32:31: [384 MB] #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ 12 Dec 2025 17:32:43: [399 MB] #2 Total number of paired peaks: 7448 \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #2 Model building with cross-correlation: Done \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #2 finished! \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #2 predicted fragment length is 186 bps \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #2 alternative fragment length(s) may be 186 bps \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #2.2 Generate R script for model : SRR1810907_peak/SRR1810907_model.r \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #3 Call peaks... \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #3 Going to call summits inside each peak ... \n",
            "INFO  @ 12 Dec 2025 17:32:48: [399 MB] #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ 12 Dec 2025 17:41:06: [1375 MB] #3 Call peaks for each chromosome... \n",
            "INFO  @ 12 Dec 2025 17:42:37: [1375 MB] #4 Write output xls file... SRR1810907_peak/SRR1810907_peaks.xls \n",
            "INFO  @ 12 Dec 2025 17:42:37: [1375 MB] #4 Write peak in narrowPeak format file... SRR1810907_peak/SRR1810907_peaks.narrowPeak \n",
            "INFO  @ 12 Dec 2025 17:42:37: [1375 MB] #4 Write summits bed file... SRR1810907_peak/SRR1810907_summits.bed \n",
            "INFO  @ 12 Dec 2025 17:42:37: [1375 MB] Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#SRR1810918(treatment), SRR1810912(control)\n",
        "%%bash\n",
        "macs3 callpeak \\\n",
        "-t SRR1810918.sorted.bam \\\n",
        "-c SRR1810912.sorted.bam \\\n",
        "-f BAM \\\n",
        "-g hs \\\n",
        "-n SRR1810918 \\\n",
        "--outdir SRR1810918_peak \\\n",
        "--keep-dup all \\\n",
        "--call-summits"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AwzSNeWOnCs0",
        "outputId": "f609adf6-d2fb-4b25-e087-4046d645fbb0"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "INFO  @ 12 Dec 2025 17:42:52: [34 MB] \n",
            "# Command line: callpeak -t SRR1810918.sorted.bam -c SRR1810912.sorted.bam -f BAM -g hs -n SRR1810918 --outdir SRR1810918_peak --keep-dup all --call-summits\n",
            "# ARGUMENTS LIST:\n",
            "# name = SRR1810918\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['SRR1810918.sorted.bam']\n",
            "# control file = ['SRR1810912.sorted.bam']\n",
            "# effective genome size = 2.91e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff = 5.00e-02\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is off\n",
            "# Paired-End mode is off\n",
            "# Searching for subpeak summits is on\n",
            " \n",
            "INFO  @ 12 Dec 2025 17:42:52: [34 MB] #1 read tag files... \n",
            "INFO  @ 12 Dec 2025 17:42:52: [34 MB] #1 read treatment tags... \n",
            "INFO  @ 12 Dec 2025 17:42:56: [65 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:42:58: [71 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:02: [73 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:06: [76 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:09: [80 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:12: [91 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:15: [91 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:19: [91 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:22: [96 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:25: [102 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:28: [105 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:33: [115 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:35: [118 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:38: [118 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:41: [127 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:45: [127 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:49: [131 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:52: [137 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:54: [137 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:43:58: [143 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:02: [146 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:05: [153 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:08: [163 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:11: [163 MB]  24000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:15: [163 MB]  25000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:18: [165 MB]  26000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:21: [171 MB]  27000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:24: [175 MB]  28000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:28: [178 MB]  29000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:32: [186 MB]  30000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:35: [186 MB]  31000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:39: [192 MB]  32000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:45: [192 MB]  33000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:48: [197 MB]  34000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:50: [242 MB] 34332852 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:44:50: [242 MB] #1.2 read input tags... \n",
            "INFO  @ 12 Dec 2025 17:44:53: [247 MB]  1000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:44:57: [253 MB]  2000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:01: [264 MB]  3000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:04: [267 MB]  4000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:07: [272 MB]  5000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:10: [272 MB]  6000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:14: [275 MB]  7000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:17: [283 MB]  8000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:20: [283 MB]  9000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:24: [285 MB]  10000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:28: [291 MB]  11000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:31: [299 MB]  12000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:34: [299 MB]  13000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:37: [302 MB]  14000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:41: [306 MB]  15000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:44: [309 MB]  16000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:47: [314 MB]  17000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:50: [319 MB]  18000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:54: [323 MB]  19000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:45:58: [326 MB]  20000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:01: [331 MB]  21000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:04: [337 MB]  22000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:07: [343 MB]  23000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:11: [346 MB]  24000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:14: [350 MB]  25000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:17: [354 MB]  26000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:20: [358 MB]  27000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:25: [364 MB]  28000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:28: [365 MB]  29000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:31: [369 MB]  30000000 reads parsed \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] 30234487 reads have been read. \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #1 tag size is determined as 50 bps \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #1 tag size = 50.0 \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #1  total tags in treatment: 34332852 \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #1  total tags in control: 30234487 \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #1 finished! \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #2 Build Peak Model... \n",
            "INFO  @ 12 Dec 2025 17:46:33: [414 MB] #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ 12 Dec 2025 17:46:52: [440 MB] #2 Total number of paired peaks: 8577 \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #2 Model building with cross-correlation: Done \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #2 finished! \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #2 predicted fragment length is 195 bps \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #2 alternative fragment length(s) may be 195 bps \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #2.2 Generate R script for model : SRR1810918_peak/SRR1810918_model.r \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #3 Call peaks... \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #3 Going to call summits inside each peak ... \n",
            "INFO  @ 12 Dec 2025 17:46:58: [440 MB] #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ 12 Dec 2025 17:55:58: [1498 MB] #3 Call peaks for each chromosome... \n",
            "INFO  @ 12 Dec 2025 17:57:31: [1498 MB] #4 Write output xls file... SRR1810918_peak/SRR1810918_peaks.xls \n",
            "INFO  @ 12 Dec 2025 17:57:31: [1498 MB] #4 Write peak in narrowPeak format file... SRR1810918_peak/SRR1810918_peaks.narrowPeak \n",
            "INFO  @ 12 Dec 2025 17:57:31: [1498 MB] #4 Write summits bed file... SRR1810918_peak/SRR1810918_summits.bed \n",
            "INFO  @ 12 Dec 2025 17:57:31: [1498 MB] Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#mount peak folder into drive\n",
        "#-r use to mount file\n",
        "!cp -r SRR1810900_peak \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp -r SRR1810907_peak \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\"\n",
        "!cp -r SRR1810918_peak \"/content/drive/MyDrive/portfolio/Chip-seq tutorial\""
      ],
      "metadata": {
        "id": "_bus74zBnPYH"
      },
      "execution_count": 7,
      "outputs": []
    }
  ]
}